pandas/_libs/algos.pyx

```diff
@@ -623,6 +623,38 @@ def pad_inplace(algos_t[:] values, const uint8_t[:] mask, limit=None):
             val = values[i]
 
 
+@cython.boundscheck(False)
```

Comment:
rather than adding new, can we simply *always* use the mask (for all dtypes)

Reply:
probably, the 1d versions of pad and backfill are only used by EA.fillna and Series.replace.

some of the EAs override fillna and use 2d algorithms. NDArrayBackedExtensionArray uses the 1d versions but should be using a 2d version (not an issue since only called columnwise, but would be fail if called directly on a 2d EA) The other EAs that use this convert to object so are not performant anyway.

not sure about Series.replace, will look some more.

Reply:
we don't support any 2D EA atm, so can leave those out for now

Reply:
done. if we are always updating and returning the mask, maybe we should return the same dtype as the original mask, or only allow boolean mask in python space and always return a boolean mask.